staging/src/k8s.io/apiserver/pkg/apis/auditregistration/validation/validation.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	audit "k8s.io/apiserver/pkg/apis/audit/validation"
+	auditconf "k8s.io/apiserver/pkg/apis/auditregistration"
+)
+
+// ValidateAuditSink validates the AuditSinks
+func ValidateAuditSink(ac *auditconf.AuditSink) field.ErrorList {
+	var allErrs field.ErrorList
+	if ac.Spec.Policy != nil {
+		allErrs = append(allErrs, audit.ValidatePolicy(ac.Spec.Policy)...)
+	}
+	allErrs = append(allErrs, ValidateWebhookBackend(ac.Spec.Backend.Webhook, field.NewPath("webhook"))...)
+	return allErrs
+}
+
+// ValidateWebhookBackend validates the webhook backend
+func ValidateWebhookBackend(b *auditconf.WebhookBackend, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if b == nil {
```

Comment:
I would expect this case in the calling func rather than here. Not sure what the convention is. Maybe we have the same elsewhere.

Reply:
I tried solving this outside the validation and it ended up a bit odd. There are some existing implementations like this, and seems to make sense here.

Reply:
I meant where `ValidateWebhookBackend` is called. Just a nit. Not important.